src/http-api.js

```diff
@@ -661,8 +687,9 @@ module.exports.MatrixHttpApi.prototype = {
                     _matrix_opts: this.opts,
                 },
                 function(err, response, body) {
-                    if (localTimeoutMs) {
-                        callbacks.clearTimeout(timeoutId);
+                    if (timeoutContext) {
+                        timeoutContext.stop();
+                        delete req.onprogres;
```

Comment:
I'm not sure whether clearing `onprogress` on a request is required or not, once it has loaded or errored.